src/binding/poem/extractor.rs

```diff
@@ -1,18 +1,21 @@
-use async_trait::async_trait;
+use crate::binding::http::to_event;
+use crate::Event;
+
 use poem_lib::error::ResponseError;
 use poem_lib::http::StatusCode;
 use poem_lib::{FromRequest, Request, RequestBody, Result};
 
-use crate::binding::http::to_event;
-use crate::Event;
-
 impl ResponseError for crate::message::Error {
     fn status(&self) -> StatusCode {
         StatusCode::BAD_REQUEST
     }
 }
 
-#[async_trait]
+// #[async_trait]
+// ^ this is causing weird lifetime issues:
+//
+// lifetime parameters or bounds on method `from_request` do not match the trait declaration
+// lifetimes do not match method in trait
```

Comment:
Can you elaborate a bit on this?

Reply:
It gives the following error:
```
error[E0195]: lifetime parameters or bounds on method `from_request` do not match the trait declaration
  --> src\binding\poem\extractor.rs:17:14
   |
17 |     async fn from_request(req: &'a Request, body: &mut RequestBody) -> Result<Self> {
   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ lifetimes do not match method in trait
```

I believe because `#[async_trait]` is not applied to poem's `FromRequest` trait.
The version of poem used previously did use it here: https://github.com/poem-web/poem/blob/v1.2.34/poem/src/web/mod.rs#L272
I didn't find the exact version where it changed, but it is gone now: https://github.com/poem-web/poem/blob/master/poem/src/web/mod.rs#L310

Let me know if you'd like me to remove the comments and or leave a better explanation.

Reply:
@jcrossley3 what do you think about this?

Reply:
I don't recall who contributed the poem impl, but if the poem version we're depending on doesn't need it, I think it's safe to omit it without any comment.

Reply:
Totally agree. @ozabalaferrera can you then please remove the comment?

Reply:
@Lazzaretti, I removed the comment and rebased to fix the sign-off. Let me know if there is anything else I should address.